File: Sway3i/src/main/java/com/sway3i/dto/StudentsInCourse/Response/EnrolledCourseResponseDTO.java
package com.sway3i.dto.StudentsInCourse.Response;

import com.sway3i.entities.Course;
import com.sway3i.entities.enums.PricingPlan;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class EnrolledCourseResponseDTO {
    private Long id;
    private LocalDate createdAt;
    private boolean isExpired;
    private Long studentId;
    private Long courseId;
    private String courseDetails;
    private String courseSubject;
    private String createdByFirstName;
    private String createdByLastName;
    private String createdByCity;
    private PricingPlan pricingPlan;
}

File: Sway3i-FrontEnd/src/app/dto/StudentInCourse/responses/enrolled-course-response-DTO.ts
import { PricingPlan } from "src/app/modal/enums/pricing-plan-enum";

export interface EnrolledCourseResponseDTO {
    id: number;
    createdAt: string; 
    isExpired: boolean;
    studentId: number;
    courseId: number;
    courseName: string;
    courseDetails: string;
    courseSubject: string;
    createdByFirstName: string;
    createdByLastName: string;
    createdByCity: string;
    pricingPlan: PricingPlan; 
    link: string;
    address: string;
  }
  